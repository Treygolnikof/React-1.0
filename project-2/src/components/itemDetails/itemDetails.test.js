import React from 'react';
import ItemDetails from './itemDetails';
import {shallow} from 'enzyme';

describe('Testing <ItemDetails/>', () => {
    const item = shallow(<ItemDetails/>);

    it('ItemDetails have rendered correctly', () => {
        expect(item).toMatchSnapshot();
    })

    it('ItemDetails state "error" is false', () => {
        expect(item.state().error).toBeFalsy();
    })

    it('ItemDetails state "loading" is true', () => {
        expect(item.state().loading).toBeTruthy();
    })

    it('ItemDetails state "item" is empty object', () => {
        expect(item.state().item).toBeNil();
    })
    
    it('Testing onItemLoaded', () => {
        item.instance().onItemLoaded();
        expect(item.state().loading).toBeFalsy();
    })

    it('Testing onError', () => {
        item.instance().onError();
        expect(item.state().error).toBeTruthy();
        expect(item.state().loading).toBeFalsy();
    })
})