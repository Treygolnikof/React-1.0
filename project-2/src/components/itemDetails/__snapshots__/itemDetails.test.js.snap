// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ItemDetails/> ItemDetails have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemDetails />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h4
        className="text-center"
      />,
      "className": "bg-white p-5",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "text-center",
      },
      "ref": null,
      "rendered": null,
      "type": "h4",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h4
          className="text-center"
        />,
        "className": "bg-white p-5",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "text-center",
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
